[quant][graphmode] Add warning for debug option for add_scalar/mul_scalar

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40414 [quant][graphmode] Enable inplace option for top level API
* #40384 [quant][graphmode][refactor] InsertQuantDeQuantHelper
* **#40383 [quant][graphmode] Add warning for debug option for add_scalar/mul_scalar**

Summary:
debug option is not supported for these cases, so we print a warning if it occurs

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22164071](https://our.internmc.facebook.com/intern/diff/D22164071)

jerryzh168 committed:
[quant][graphmode] Add warning for debug option for add_scalar/mul_scalar

Summary:
debug option is not supported for these cases, so we print a warning if it occurs

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo reviewed (approved):
> we don't check the numerical consistency for add_scalar since it's not supported

just curious, what's the context on why it's not supported?

jerryzh168 commented:
> > we don't check the numerical consistency for add_scalar since it's not supported
> 
> just curious, what's the context on why it's not supported?

oh, the reason is written in the comments

jerryzh168 commented on torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp:
```diff
@@ -1024,6 +1024,16 @@ void InsertQuantDeQuantHelper::propagateQuantizationOps(Block* block) {
         }
       }
+    } else if (isPropagateQuantNode(n)) {
+      // Print warning for add_scalar/mul_scalar when debug is enabled
```
@vkuzo

jerryzh168 committed:
Update on "[quant][graphmode] Add warning for debug option for add_scalar/mul_scalar"


Summary:
debug option is not supported for these cases, so we print a warning if it occurs

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22164071](https://our.internmc.facebook.com/intern/diff/D22164071)

[ghstack-poisoned]

vkuzo commented:
> > > we don't check the numerical consistency for add_scalar since it's not supported
> > 
> > 
> > just curious, what's the context on why it's not supported?
> 
> oh, the reason is written in the comments

thanks, wasn't sure if it was that comment or something else, got it

jerryzh168 committed:
Update on "[quant][graphmode] Add warning for debug option for add_scalar/mul_scalar"


Summary:
debug option is not supported for these cases, so we print a warning if it occurs

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22164071](https://our.internmc.facebook.com/intern/diff/D22164071)

[ghstack-poisoned]
